src/modules/home/components/accommodation-list.tsx

```diff
@@ -0,0 +1,76 @@
+import Grid2 from '@mui/material/Grid2';
+import CardImage from '#/assets/images/card-temp-image.jpg';
+import React, { useEffect, useRef, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { AccommodationCard } from './accommodation-card';
+import { useListAccommodationsQuery } from '../api/list-accommodations.api';
+import { SkeletonList } from './skeleton-list';
+import { CardSkeleton } from './card-skeleton';
+import { calculateInitialLimit } from '../utils/calculate-initial-limit.helper';
+import { useInView } from 'react-intersection-observer';
+import { useListAccommodationQueryParams } from '#/shared/hooks/list-accommodations-query-params.hook';
+
+export function AccommodationList() {
+  const [limit, setLimit] = useState(20);
+  const { ref: bottomOfPageRef, inView: isBottomOfPageInView } = useInView();
+  const { validatedQueryParams } = useListAccommodationQueryParams();
+  const cardRef = useRef<HTMLDivElement>(null);
+
+  const { fetchNextPage, hasNextPage, isFetchingNextPage, status, data } = useListAccommodationsQuery(
+    limit,
+    validatedQueryParams
+  );
+
+  useEffect(() => {
+    const initialNumberOfCards = calculateInitialLimit(cardRef.current, limit);
+
+    setLimit(initialNumberOfCards);
+  }, [limit]);
+
+  useEffect(() => {
```

Comment:
Same here, better to try to move to separate hook this logic or try to combine it with the useEffect above.

Reply:
> Same here, better to try to move to separate hook this logic or try to combine it with the useEffect above.

Created separate hooks for both